libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1015,8 +1016,21 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         setRoutesInfo: SetRoutes,
         callback: RoutesSetCallback? = null,
     ) {
-        rerouteController?.interrupt()
-        restartRouteScope()
+        logD(LOG_CATEGORY) {
+            "setting routes; reason: ${setRoutesInfo.mapToReason()}; IDs: ${routes.map { it.id }}"
+        }
+        when (setRoutesInfo) {
+            SetRoutes.CleanUp,
+            is SetRoutes.NewRoutes,
+            SetRoutes.Reroute -> {
+                rerouteController?.interrupt()
+            }
+            is SetRoutes.RefreshRoutes,
+            is SetRoutes.Alternatives -> {
+                // do not interrupt reroute when primary route has not changed
+            }
```

Comment:
Was thinking about this for a little, It's worth adding definitions to `SetRoutes` because logic depends on the definition. Thinking about scenarios where someone should be calling `NewRoutes` but they decide to call `SetRoutes.Alternatives`.

```

    /** 
     * Triggered when the alternative routes change and the primary route has remained the same.
     */
    internal data class Alternatives(
        val legIndex: Int,
    ) : SetRoutes()
```